source/src/cfg/cs_Boards.c

```diff
@@ -213,3 +223,77 @@ cs_ret_code_t configure_board(boards_config_t* config) {
 
 	return ERR_SUCCESS;
 }
+
+cs_ret_code_t configure_board_from_uicr(const cs_uicr_data_t* uicrData, boards_config_t* config) {
+	init(config);
+	config->hardwareBoard = uicrData->board;
+
+	switch (uicrData->productRegionFamily.fields.productType) {
+		case PRODUCT_DEV_BOARD:
+			asPca10040(config);
+			return ERR_SUCCESS;
+		case PRODUCT_CROWNSTONE_PLUG_ZERO: {
+			switch (uicrData->majorMinorPatch.fields.major) {
+				case 0:
+					asACR01B2C(config);
+					return ERR_SUCCESS;
+				case 1:
+					switch (uicrData->majorMinorPatch.fields.minor) {
+						case 0:
+							asACR01B2C(config);
+							return ERR_SUCCESS;
+						case 1:
```

Comment:
Add `[[fallthrough]];`, will anticipate newer compilers which complain about more things.

Reply:
Doesn't seem to work for .c files.

Reply:
Then use `__attribute__ ((fallthrough));`